ffi.c

```diff
@@ -435,7 +435,21 @@ EXP K deref(K x) {
   return r;
 }
 
-#define N 9
+Z K1(cv){
+  V *v;
+  H t;
+  if (xt==-KS) {
+    t = KI;
+  } else {
+    P(xt||xn!=2||xK[0]->t!=-KC,krr("type"));
```

Comment:
use simple if and direct access to members. `xK->kK(x)` to match rest of the file